### PR TITLE
[FEATURE] Snowflake - Key-Pair auth updates from `0.18.x`

### PR DESCRIPTION
**Pull request by Kilo59**
Applying various Snowflake updates from the `0.18.x` branch to `1.0`

1. Add support for Key-Pair auth
    a. https://github.com/great-expectations/great_expectations/pull/10031
    b. https://github.com/great-expectations/great_expectations/pull/10046

### CODE REVIEW

File: great_expectations/compatibility/not_imported.py
Comment (Kilo59) [inline great_expectations/compatibility/not_imported.py:44]: Move these changes to a different PR

File: tasks.py
Comment (Kilo59) [inline tasks.py]: This change prevents the `marker-converage` static analysis check from erroring on a warning without changing anything about our dependencies.

Comment (Kilo59) [reply tasks.py]: Filtering in pyproject.toml instead

File: tests/datasource/fluent/integration/test_connections.py
Comment (Kilo59) [inline tests/datasource/fluent/integration/test_connections.py:1]: The changes in this file + the pyproject.toml warning filter are what should unblock the CI.

The warning is only emitted when the latest `snowflake-sqlalchemy` is installed alongside `sqlalchemy 1`.
There is no warning when `sqlalchemy 2` is installed alongside the latest snowflake dep.

File: pyproject.toml
Comment (Kilo59) [inline pyproject.toml]:
```suggestion
    # snowflake
    # Warning is emitted when snowflake-sqlalchemy 1.6 is installed alongside sqlalchemy 1
    "module: The GenericFunction 'flatten' is already registered and is going to be overridden:sqlalchemy.exc.SAWarning"
```

File: great_expectations/datasource/fluent/sql_datasource.py
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py]:
```suggestion
```

**Review by tyler-hoffman — APPROVED** [review]
Looks reasonable. For most of this, I just checked that it matches the 0.18 branch.

File: great_expectations/datasource/fluent/sql_datasource.py
Comment (Kilo59) [inline great_expectations/datasource/fluent/sql_datasource.py:121]: @anthonyburdi This is how I'm addressing the `TRY003` warnings.
https://docs.astral.sh/ruff/rules/raise-vanilla-args/

File: tests/integration/cloud/end_to_end/test_datacontext_v1_api.py
Comment (Kilo59) [inline tests/integration/cloud/end_to_end/test_datacontext_v1_api.py:33]: @tyler-hoffman @joshua-stauffer XFAILing this test

File: tests/datasource/fluent/integration/test_connections.py
Comment (Kilo59) [reply tests/datasource/fluent/integration/test_connections.py:1]: cc @billdirks

File: great_expectations/datasource/fluent/snowflake_datasource.py
Comment (Kilo59) [inline great_expectations/datasource/fluent/snowflake_datasource.py:496]: @roblim adding this to 1.0

File: tests/integration/cloud/end_to_end/test_datacontext_v1_api.py
Comment (joshua-stauffer) [reply tests/integration/cloud/end_to_end/test_datacontext_v1_api.py:33]: i have the fix for this [here](https://github.com/great-expectations/great_expectations/pull/10093), currently blocked by ci

File: great_expectations/datasource/fluent/sql_datasource.py
Comment (anthonyburdi) [reply great_expectations/datasource/fluent/sql_datasource.py:121]: @Kilo59 thanks for the ping. One question: Should there be validation based on the docstring? 
```
"""Must provide a `cause`, `addendum`, or both."""
```
I'm thinking no, because we wouldn't want the exception to raise an exception.

Comment (Kilo59) [reply great_expectations/datasource/fluent/sql_datasource.py:121]: It's enforced via the type-checker and the declared `overloads`.

**Pull request merged**
